integration_tests/__tests__/custom-reporters-test.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+const runJest = require('../runJest');
+const skipOnWindows = require('skipOnWindows');
+
+describe('Custom Reporters Integration', () => {
+  skipOnWindows.suite();
+
+  test('valid string format for adding reporters', () => {
+    const reporterConfig = {
+      'reporters': [
+        '<rootDir>/reporters/TestReporter.js',
+      ],
+    };
+
+    const {status} = runJest('custom_reporters', [
+      '--config',
+      JSON.stringify(reporterConfig),
+      'add-test.js',
+    ]);
+
+    expect(status).toBe(0);
+  });
+
+  test('valid array format for adding reporters', () => {
+    const reporterConfig = {
+      'reporters': [
+        ['<rootDir>/reporters/TestReporter.js', {'Dmitrii Abramov': 'Awesome'}],
+      ],
+    };
+
+    const {status} = runJest('custom_reporters', [
+      '--config',
+      JSON.stringify(reporterConfig),
+      'add-test.js',
+    ]);
+
+    expect(status).toBe(0);
+  });
+
+  test('invalid format for adding reporters', () => {
+    const reporterConfig = {
+      'reporters': [
+        [3243242],
+      ],
+    };
+
+    const {status, stderr} = runJest('custom_reporters', [
+      '--config',
+      JSON.stringify(reporterConfig),
+      'add-test.js',
+    ]);
+
+    expect(stderr).toMatch('Expected reporterPath for reporter');
+    expect(status).toBe(1);
+  });
+  
+  test('TestReporter with all tests passing', () => {
+    const {
+      stdout,
+      status,
+      stderr,
+    } = runJest('custom_reporters', ['add-test.js']);
+    let parsedJSON;
+
+    try {
+      parsedJSON = JSON.parse(stdout);
+    } catch (error) {
+      throw new Error(
+        'Failed to parse JSON, Check the Output of TestReporter'
+      );
+    }
+
+    const {onRunComplete, onRunStart, onTestResult, onTestStart} = parsedJSON;
+
+    expect(status).toBe(0);
+    expect(stderr.trim()).toBeFalsy();
```

Comment:
is that supposed to be an empty string?
if yes then i'd prefer it to be explicit `.toBe('')`